runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -357,6 +357,50 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
     return new DataflowRunner(dataflowOptions);
   }
 
+  /**
+   * Get a default value for Google Cloud region according to
+   * https://cloud.google.com/compute/docs/gcloud-compute/#default-properties. If no other default
+   * can be found, returns "us-central1".
+   */
+  static String getDefaultGcpRegion() {
+    String environmentRegion = System.getenv("CLOUDSDK_COMPUTE_REGION");
+    if (environmentRegion != null && !environmentRegion.isEmpty()) {
+      LOG.info("Using default GCP region {} from $CLOUDSDK_COMPUTE_REGION", environmentRegion);
+      return environmentRegion;
+    }
+    try {
+      ProcessBuilder pb =
+          new ProcessBuilder(Arrays.asList("gcloud", "config", "get-value", "compute/region"));
+      Process process = pb.start();
+      BufferedReader reader =
+          new BufferedReader(
+              new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
+      BufferedReader errorReader =
+          new BufferedReader(
+              new InputStreamReader(process.getErrorStream(), StandardCharsets.UTF_8));
+      process.waitFor(1, TimeUnit.SECONDS);
```

Comment:
- Is 1 second enough?
- Should we check the return value of this call? (IIUC it returns false for timeout.)

Reply:
> Is 1 second enough?

Tests on my machine (in debug mode) indicated it took on average around 0.25 seconds, including reading the output stream after. I have bumped the timeout to 2s just to make double sure.

> Should we check the return value of this call?

Not sure it matters, since I think it should interrupt the process after the time limit has passed, but it couldn't hurt to check anyway.